tests/libvmops/run.go

```diff
@@ -39,65 +40,57 @@ import (
 	"kubevirt.io/client-go/kubecli"
 )
 
-func RunVMIAndExpectLaunch(vmi *v1.VirtualMachineInstance, timeout int) *v1.VirtualMachineInstance {
-	vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
-	Expect(err).ToNot(HaveOccurred())
-	By("Waiting until the VirtualMachineInstance will start")
-	return libwait.WaitForVMIPhase(vmi,
-		[]v1.VirtualMachineInstancePhase{v1.Running},
-		libwait.WithTimeout(timeout),
-	)
-}
+func RunVMAndExpectLaunchWithRunStrategy(virtClient kubecli.KubevirtClient, vm *v1.VirtualMachine, runStrategy v1.VirtualMachineRunStrategy) *v1.VirtualMachine {
+	By("Starting the VirtualMachine")
+	vm, err := updateVMRunningStatus(virtClient, vm, &runStrategy)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
 
-func RunVMIAndExpectLaunchIgnoreWarnings(vmi *v1.VirtualMachineInstance, timeout int) *v1.VirtualMachineInstance {
-	vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
-	Expect(err).ToNot(HaveOccurred())
-	By("Waiting until the VirtualMachineInstance will start")
-	return libwait.WaitForSuccessfulVMIStart(vmi,
-		libwait.WithFailOnWarnings(false),
-		libwait.WithTimeout(timeout),
-	)
-}
+	By("Waiting for VMI to be running")
+	EventuallyWithOffset(1, ThisVMIWith(vm.Namespace, vm.Name), 300*time.Second, 1*time.Second).Should(BeRunning())
 
-func RunVMIAndExpectScheduling(vmi *v1.VirtualMachineInstance, timeout int) *v1.VirtualMachineInstance {
-	wp := watcher.WarningsPolicy{FailOnWarnings: true}
-	return RunVMIAndExpectSchedulingWithWarningPolicy(vmi, timeout, wp)
-}
+	By("Waiting for VM to be ready")
+	EventuallyWithOffset(1, ThisVM(vm), 360*time.Second, 1*time.Second).Should(BeReady())
 
-func RunVMIAndExpectSchedulingWithWarningPolicy(vmi *v1.VirtualMachineInstance, timeout int, wp watcher.WarningsPolicy) *v1.VirtualMachineInstance {
-	vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
-	Expect(err).ToNot(HaveOccurred())
-	By("Waiting until the VirtualMachineInstance will be scheduled")
-	return libwait.WaitForVMIPhase(vmi,
-		[]v1.VirtualMachineInstancePhase{v1.Scheduling, v1.Scheduled, v1.Running},
-		libwait.WithWarningsPolicy(&wp),
-		libwait.WithTimeout(timeout),
-	)
+	vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+	return vm
 }
 
-func RunVMAndExpectLaunchWithRunStrategy(virtClient kubecli.KubevirtClient, vm *v1.VirtualMachine, runStrategy v1.VirtualMachineRunStrategy) *v1.VirtualMachine {
-	By("Starting the VirtualMachine")
+func updateVMRunningStatus(virtClient kubecli.KubevirtClient, vm *v1.VirtualMachine, runStrategy *v1.VirtualMachineRunStrategy) (*v1.VirtualMachine, error) {
```

Comment:
But in this case this function is used as a generic helper in both Start or Stop depending on the passed run strategy. Update seems more appropriate because we are updating (or patching) the running strategy, wdyt?

Reply:
Well, there are verbs in our client that expresses start and stop that are a better wrapper IMO to this action here. It replaces the need to perform to operations: Preparing the updated spec and then sending a command to update/patch the object.

What would be the reason to prefer a direct update/patch over a verb in this case?